File: ThreeTeleports/solve/cpp/ThreeTeleportsSolver.cpp
#include <fstream>
#include <topcoder>
#include "ThreeTeleports.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int xMe; tc::read(ifs, xMe); tc::next(ifs);
    int yMe; tc::read(ifs, yMe); tc::next(ifs);
    int xHome; tc::read(ifs, xHome); tc::next(ifs);
    int yHome; tc::read(ifs, yHome); tc::next(ifs);
    vector<string> teleports; tc::read(ifs, teleports);
    ifs.close();

    std::ofstream ofs(argv[2]);
    ThreeTeleports solver;
    tc::show(ofs, solver.shortestDistance(xMe, yMe, xHome, yHome, teleports));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}

File: ThreeTeleports/solve/java/ThreeTeleportsSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class ThreeTeleportsSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        int xMe = (Integer) reader.next(Integer.class);
        reader.next();
        
        int yMe = (Integer) reader.next(Integer.class);
        reader.next();
        
        int xHome = (Integer) reader.next(Integer.class);
        reader.next();
        
        int yHome = (Integer) reader.next(Integer.class);
        reader.next();
        
        List<String> teleportsBoxed = (List<String>) reader.next(new TypeRef<List<String>>(){}.getType());
        String[] teleports = new String[teleportsBoxed.size()];
        for (int _i = 0; _i < teleportsBoxed.size(); ++_i)
            teleports[_i] = teleportsBoxed.get(_i);
        reader.close();

        ThreeTeleports solver = new ThreeTeleports();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.shortestDistance(xMe, yMe, xHome, yHome, teleports));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}
